pkg/controller/injector/k8s_binding_injector_test.go

```diff
@@ -29,6 +30,123 @@ import (
 	kapi "k8s.io/kubernetes/pkg/api"
 )
 
+func TestCreateSerializedSecret(t *testing.T) {
+	cases := []struct {
+		name   string
+		cred   *brokerapi.Credential
+		secret *v1.Secret
+	}{
+		{
+			name: "string type",
+			cred: &brokerapi.Credential{
+				"Hostname": "host",
+				"Port":     "123",
+				"Username": "user",
+				"Password": "password!@#!@#!0)",
+			},
+			secret: &v1.Secret{
+				Data: map[string][]byte{
+					"Hostname": []byte("host"),
+					"Port":     []byte("123"),
+					"Username": []byte("user"),
+					"Password": []byte("password!@#!@#!0)"),
+				},
+			},
+		},
+		{
+			name: "float type",
+			cred: &brokerapi.Credential{
+				"Hostname": "host",
+				"Port":     "123",
+				"Username": "user",
+				"Password": 1.23,
+			},
+			secret: &v1.Secret{
+				Data: map[string][]byte{
+					"Hostname": []byte("host"),
+					"Port":     []byte("123"),
+					"Username": []byte("user"),
+					"Password": []byte("1.23"),
+				},
+			},
+		},
+		{
+			name: "int type",
+			cred: &brokerapi.Credential{
+				"Hostname": "host",
+				"Port":     123,
+				"Username": "user",
+				"Password": 1,
+			},
+			secret: &v1.Secret{
+				Data: map[string][]byte{
+					"Hostname": []byte("host"),
+					"Port":     []byte("123"),
+					"Username": []byte("user"),
+					"Password": []byte("1"),
+				},
+			},
+		},
+		{
+			name: "slice type",
+			cred: &brokerapi.Credential{
+				"Hostname": "host",
+				"Port":     "123",
+				"Username": "user",
+				"Password": []string{"one", "two"},
+			},
+			secret: &v1.Secret{
+				Data: map[string][]byte{
+					"Hostname": []byte("host"),
+					"Port":     []byte("123"),
+					"Username": []byte("user"),
+					"Password": []byte(`["one","two"]`),
+				},
+			},
+		},
+		{
+			name: "map type",
+			cred: &brokerapi.Credential{
+				"Hostname": "host",
+				"Port":     "123",
+				"Username": "user",
+				"Password": map[string]int{"one": 1, "two": 2},
```

Comment:
What is the purpose of this case? Where would we have maps? Is there a better field that would actually be a map?